assignment3.txt

```diff
@@ -0,0 +1,6 @@
+Jack Rosen
+1. This is valid but it is assumed that the variable i has already been initialized.
+2. This is an infinite loop since i is always below 10 because it only gets smaller than 0 and never reaches 10.
+3. i.O(n^2)
+ii. O(n^2 + n)
```

Comment:
Not quite! The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2.

Lastly, in big notation, you disregard the lower terms, so either way, your answer would be wrong because it included "+ n"